pandas/core/sorting.py

```diff
@@ -517,16 +518,22 @@ def get_flattened_list(
     return [tuple(array) for array in arrays.values()]
 
 
-def get_indexer_dict(label_list, keys):
+def get_indexer_dict(
+    label_list: List[ABCIndex], keys: List[np.ndarray]
+) -> Dict[Union[str, Tuple], np.ndarray]:
```

Comment:
The keys can also be integers e.g. `pd.DataFrame({"a": [1], "b": [0]}).groupby('a').indices` or indeed, I think any hashable. Can use pandas._typing.Label here.

Reply:
Thanks